File: Csharp/CodeWars/src/Resolutions/StrongNumberSpecialNumbersSeries2.cs
namespace CodeWars.Resolutions;

public class StrongNumberSpecialNumbersSeries2
{
    public static string StrongNumber(int number) => number
         .ToString()
         .Sum(x =>
            Convert.ToInt32(
                CalculateFactorial((int)char.GetNumericValue(x))
            )) == number
        ? "STRONG!!!!"
        : "Not Strong !!";

    private static int CalculateFactorial(int N)
    {
        if (N == 0 || N == 1)
            return 1;

        int factorial = 1;
        for (int i = N; i > 1; i--)
            factorial *= i;

        return factorial;
    }
}

File: TypeScript/Codewars/src/resolutions/Strong-Number-Special-Numbers-Series-2.ts
export const strongNumber = (num: number): string => {
	const result = Array.from(num.toString())
		.map(Number)
		.reduce((prev, curr) => {
			return prev + factorial(curr);
		}, 0);

	return result === num ? "STRONG!!!!" : "Not Strong !!";
};

function factorial(n: number) {
	return n <= 1 ? 1 : n * factorial(n - 1);
}
